src/subscript/welltest_dpds/welltest_dpds.py

```diff
@@ -0,0 +1,475 @@
+
+"""
+Script is a rewrite of a legacy script originally developed and improved by:
+ * Jon Sætrom
+ * Bjørn Kåre Hegstad
+ * Cecile Otterlei
+ * Hodjat Moradi
+
+AUTHOR:
+ * Eivind Smørgrav
+
+To do:
+ * Support pseudo pressure vs time relevant for gas and gas condensate fields.
+ * Add cli option to output other vectors in csv file
+
+:meta private:
+"""
+
+CATEGORY = "modelling.reservoir"
+
+EXAMPLES = """
+.. code-block:: console
+
+ FORWARD_MODEL WELLTEST_DPDS(<ECLBASE>, <WELLNAME>=DST_WELL)
+
+ FORWARD_MODEL  WELLTEST_DPDS(<ECLBASE>, <WELLNAME>=OP_1, <PHASE>=GAS,
+    <BUILDUP_NR>=1, <OUTPUTDIRECTORY>=dst, <OUTFILESSUFIX>=OP_1_1)
+
+"""
+
+
+def get_parser():
+    """
+    Define the argparse parser
+
+    Returns:
+        parser (argparse.ArgumentParser)
+    """
+    parser = argparse.ArgumentParser(
+        description=DESCRIPTION,
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+
+    parser.add_argument(
+        "eclcase",
+        type=str,
+        help="Eclipse case to extract from",
+    )
+    parser.add_argument(
+        "wellname",
+        type=str,
+        help="Name of well to extract results from",
+    )
+    parser.add_argument(
+        "--outfilessuffix",
+        type=str,
+        help='Suffix to be added to result files. Default: ""',
+        default="",
+    )
+    parser.add_argument(
+        "-n",
+        "--buildup_nr",
+        type=int,
+        help="Buildup nr, eg which buildup to extract. Counting from 1. Default: 1 ",
+        default=1,
+    )
+    parser.add_argument(
+        "-o",
+        "--outputdirectory",
+        type=str,
+        help="Directory to put the output files. Detault: .",
+        default=".",
+    )
+    parser.add_argument(
+        "--phase",
+        type=str,
+        choices=["OIL", "GAS"],
+        help="Main fluid phase in test (OIL/GAS). Default: OIL",
+        default="OIL",
+        required=False,
+    )
+    return parser
+
+
+def get_summary_vec(summary, key, required=True):
+    """
+    Read vector corresponding to key from summary instance
+
+    Args:
+       summary: (ecl.summary.EclSum)
+       key: (str)
+       required: (bool)
+
+    Returns:
+        vec  : np.array
+
+    """
+
+    try:
+        return summary.numpy_vector(key)
+    except KeyError:
+        print("No such key in summary file:" + key)
+        if required:
+            raise
+    return np.array([])
+
+
+def get_buildup_indices(rates):
+    """
+    Go through the simulated rate and identify bu periods, as defined by zero flow.
+
+    Returns:
+    buildup_incices     : list of indices associated with start of buildups
+    buildup_end_incices : list of indices associated with end of buildups
+
+    Args:
+       rates: np.array
+
+    Returns:
+       buildup_indices (list)
+       buildup_end_indices (list)
+
+    """
+
+    buildup_indices = []
+    buildup_end_indices = []
+
+    last = 0
+    for i, rate in enumerate(rates):
+        if rate == 0 and last > 0.0:
```

Comment:
`rate == 0` is a floating point comparison. Might be safer to do `np.isclose(rate, 0)`